### ISSUE TITLE
Dio.js is failing to provide the actual cause of the bugs in some cases

### ISSUE THREAD

**Issue by rosenfeld** [thysultan/dio.js#69]
I failed to reproduce a simple code to demonstrate the issue, but if you still have access to that sequel-roda-webpack-starter project in GitLab could you please give the "dio-error-bug" branch a try?

Try to understand where the error comes from without looking at the last commit in that branch.

Thanks in advance.

**Comment by thysultan** [comment]
Is this related to error boundaries? Can you post the logged stack trace?

**Comment by rosenfeld** [comment]
No, I've thought about that too but even if I disable the error boundary component I get the same error:

<pre>
TypeError: Cannot read property 'constructor' of null
    at getLifecycleRefs (vendors~load-router~vendor.b05197041d95edcadf9f.js:1806)
    at commitOwnerRefs (vendors~load-router~vendor.b05197041d95edcadf9f.js:2514)
    at commitUnmountElementChildren (vendors~load-router~vendor.b05197041d95edcadf9f.js:2414)
    at commitUnmountElementChildren (vendors~load-router~vendor.b05197041d95edcadf9f.js:2408)
    at commitUnmountElementChildren (vendors~load-router~vendor.b05197041d95edcadf9f.js:2408)
    at commitUnmountElement (vendors~load-router~vendor.b05197041d95edcadf9f.js:2379)
    at commitMountElementReplacement (vendors~load-router~vendor.b05197041d95edcadf9f.js:2358)
    at clearErrorBoundary (vendors~load-router~vendor.b05197041d95edcadf9f.js:1219)
    at propagateErrorBoundary (vendors~load-router~vendor.b05197041d95edcadf9f.js:1249)
    at recoverErrorBoundary (vendors~load-router~vendor.b05197041d95edcadf9f.js:1209) Exception {error: TypeError: Cannot read property 'constructor' of null
    at getLifecycleRefs (http://localhost:4000…, origin: "callback", bubbles: true, composed: false, Symbol(dio.Element): null}
(index):1 Uncaught (in promise) 
</pre>

However, sometimes I get some other errors that are even worse than that to try to understand. My hope is that once this is fixed the other errors could be caused by the same issue.

If I disable the error boundary the error is reported like this:

<pre>
Uncaught TypeError: Cannot read property 'host' of null
    at Exception.get (vendors~load-router~vendor.b05197041d95edcadf9f.js:1129)
    at Exception.get (vendors~load-router~vendor.b05197041d95edcadf9f.js:1117)
    at listener (vendors~setup-bugsnag.185a7ca33bd47b75f65e.js:1820)
get @ vendors~load-router~vendor.b05197041d95edcadf9f.js:1129
get @ vendors~load-router~vendor.b05197041d95edcadf9f.js:1117
listener @ vendors~setup-bugsnag.185a7ca33bd47b75f65e.js:1820
(index):1 Uncaught (in promise) 
</pre>

**Comment by thysultan** [comment]
@rosenfeld Where can i find the setup/install instructions.

**Comment by rosenfeld** [comment]
Give me a few minutes and I'll add some commits to make it easier for you to run the project without the database.

**Comment by rosenfeld** [comment]
I've just added a new file QUICK_FRONTEND_SETUP.md with the instructions to that branch. Hopefully that will be enough.

**Comment by thysultan** [comment]
How do i turn of the mangler/minifier. My stack traces is not as clear as yours.

**Comment by rosenfeld** [comment]
Sorry, forgot to comment that line, I'll do and commit to that branch

**Comment by rosenfeld** [comment]
Done, a git pull should fix it but you'll have to restart webpack

**Comment by thysultan** [comment]
Found the problem: It's trying to remove a ref of an element that failed to render(ergo it wouldn't have a live ref). For example [https://github.com/thysultan/dio.js/blob/master/src/Core/Commit.js#L151](https://github.com/thysultan/dio.js/blob/master/src/Core/Commit.js#L151) putting a `if (!element.active) return` fixes the issue. Will release a patch for this.

...Though after that i get the error.

```
TypeError: Cannot read property 'action' of undefined
    at Link.setInitialState (load-router.7b79c0d0db8027144bd3.js:180)
    at new BaseComponentCommon (load-router.7b79c0d0db8027144bd3.js:59)
    at new BaseComponent (load-router.7b79c0d0db8027144bd3.js:161)
    at new Link (load-router.7b79c0d0db8027144bd3.js:834)
```

Which i can only assume is not related to DIO specifically.

**Comment by rosenfeld** [comment]
Yes, I caused that error in one of the commits to illustrate the problem. This is what I was expecting to get in order to being able to find and fix the bugs I have sometimes. Before that fix I'd have to switch to React to understand where the bug was coming from.

**Comment by rosenfeld** [comment]
Thanks for the fix. Once it's released I'll try it out and see if it also fixes the other cases like that I have in my own application.

**Comment by thysultan** [comment]
Patched in 9.1.1

**Comment by rosenfeld** [comment]
Thanks, I'll try it out :)
